src/frontc/cabs2cil.ml

```diff
@@ -1181,6 +1174,7 @@ class registerLabelsVisitor = object
 
   method! vstmt s =
     currentLoc := convLoc (C.get_statementloc s);
+    (* don't touch currentExpLoc! *)
```

Comment:
Can you elaborate on why one should not touch it here?

Reply:
I just extended the comments related to this. There's no deep reason other than if you change it here, it also needs to be reset after using the visitor in `doDecl`. The comment is mainly here as a reminder not to touch it here without resetting it afterwards and possibly breaking something.